python/whylogs/api/writer/whylabs.py

```diff
@@ -679,10 +717,26 @@ def write(self, file: Writable, **kwargs: Any) -> Tuple[bool, str]:
                 )
 
             dataset_timestamp_epoch = int(stamp * 1000)
-            response = self._do_upload(
-                dataset_timestamp=dataset_timestamp_epoch,
-                profile_file=tmp_file,
-            )
+            if self._transaction_id is not None:
```

Comment:
we have different codepaths for feature weights, estimation result, or segmented result sets. If I'm understanding this correctly, transactional support won't work for these cases, right?

Reply:
Segmented profiles call back to `write()` for each segment, so I think those would work. I think segmented _reference_ profiles and the other ones you mention would need additional transactional client API endpoints, or perhaps make transaction ID an optional parameter on those endpoints... Right now, writing those things when a transaction is open works, but they're not included in the transaction. Another option would be to issue a warning or error on writing non-transactable things when a transaction is open. Or maybe have `start_transaction()` return an ID that gets passed to `write()` as an optional keyword argument. I'd prefer not do add more transaction-specific methods to the `WhyLabsWriter` without a major refactoring.

Reply:
other than seg reference profiles, the other ones are very lightweight, so maybe for now we could just emit a warning for when they are being uploaded while a transaction is open?

Reply:
Verified it throws on failure.